### ISSUE TITLE
Type for `InView` `onChange` prop is overriden by union

### ISSUE THREAD

**Issue by etripier** [thebuilder/react-intersection-observer#202]
The following typedef is given for the `InView` component props:

```typescript
export declare type PlainChildrenProps = IntersectionOptions & {
    children: React.ReactNode;
    /**
     * Render the wrapping element as this element.
     * @default `'div'`
     */
    as?: React.ReactType<any>;
    /**
     * Element tag to use for the wrapping component
     * @deprecated Replace with the 'as' prop
     */
    tag?: React.ReactType<any>;
    /** Call this function whenever the in view state changes */
    onChange?: (inView: boolean, entry: IntersectionObserverEntry) => void;
} & React.HTMLProps<HTMLDivElement>;
```

The union with `React.HTMLProps<HTMLDivElement>` overrides the `onChange` typedef so that the end result is `((inView: boolean, entry: IntersectionObserverEntry) => void) & ((event: FormEvent<HTMLDivElement>) => void)`. This makes it impossible to pass a function to `onChange` that satisfies the constraints.

The problem can be fixed by doing a pick on `React.HTMLProps<HTMLDivElement>` to avoid the pre-defined props. Also, I suspect you may want to use a generic, just in case the component is rendered as something other than a `div`.

**Comment by thebuilder** [comment]
Would you like to do a pull request with this? 😃 Haven't tried to use Pick before.
I was vary of using the `HTMLDivElement` directly, but TypeScript was complaining otherwise. Would the generic be `HTMLElement`?

**Comment by thebuilder** [comment]
@etripier does the change i made in #205 make sense? 🤔

**Comment by thebuilder** [comment]
Fixed in #205
